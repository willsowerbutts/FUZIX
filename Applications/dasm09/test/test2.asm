;
; 6309 assembler/disassembler test file (12 Aug 1999)
;

     adcd #$0101
     adcd <$01
     adcd ,x
     adcd $0101

     adde #$01
     adde <$01
     adde ,x
     adde $0101

     addf #$01
     addf <$01
     addf ,x
     addf $0101

     addw #$0101
     addw <$01
     addw ,x
     addw $0101

     aim #$01,<$01
     aim #$01,$0101
     aim #$01,,x

     addd #$0101
     addd <$01
     addd ,x
     addd $0101

     asld

     asrd

     bitmd #$01

     clrd
     clre
     clrf
     clrw

     cmpe #$01
     cmpe <$01
     cmpe ,x
     cmpe $0101

     cmpf #$01
     cmpf <$01
     cmpf ,x
     cmpf $0101

     cmpw #$0101
     cmpw <$01
     cmpw ,x
     cmpw $0101

     comd
     come
     comf
     comw

     decd
     dece
     decf
     decw

     divd #$01
     divd <$01
     divd ,x
     divd $0101

     divq #$0101
     divq <$01
     divq ,x
     divq $0101

     eim #$01,<$01
     eim #$01,$0101
     eim #$01,,x

     eord #$0101
     eord <$01
     eord ,x
     eord $0101

;tfr exg

     incd
     ince
     incf
     incw

     lde #$01
     lde <$01
     lde ,x
     lde $0101

     ldf #$01
     ldf <$01
     ldf ,x
     ldf $0101

     ldq #$01010101
     ldq <$01
     ldq ,x
     ldq $0101

     ldw #$0101
     ldw <$01
     ldw ,x
     ldw $0101

     ldmd #$01

     lsrd
     lsrw

     muld #$0101
     muld <$01
     muld ,x
     muld $0101

     negd

     oim #$01,<$01
     oim #$01,$0101
     oim #$01,,x

     ord #$0101
     ord <$01
     ord ,x
     ord $0101

     pshsw
     pshuw

     pulsw
     puluw

     rold
     rolw

     rord
     rorw

     sbcd #$0101
     sbcd <$01
     sbcd ,x
     sbcd $0101

     sexw

     ste <$01
     ste $0101
     ste ,x

     stf <$01
     stf $0101
     stf ,x

     stq <$01
     stq $0101
     stq ,x

     stw <$01
     stw $0101
     stw ,x

     sube #$01
     sube <$01
     sube ,x
     sube $0101

     subf #$01
     subf <$01
     subf ,x
     subf $0101

     subw #$0101
     subw <$01
     subw ,x
     subw $0101

     tim #$01,<$01
     tim #$01,$0101
     tim #$01,,x

     tstd
     tste
     tstf
     tstw

     band cc.0,<$01.0
     band cc.1,<$01.1
     band cc.2,<$01.2
     band cc.3,<$01.3
     band cc.4,<$01.4
     band cc.5,<$01.5
     band cc.6,<$01.6
     band cc.7,<$01.7
     band cc.7,<$01.0
     band cc.0,<$01.7

     band a.0,<$01.0
     band a.1,<$01.1
     band a.2,<$01.2
     band a.3,<$01.3
     band a.4,<$01.4
     band a.5,<$01.5
     band a.6,<$01.6
     band a.7,<$01.7
     band a.7,<$01.0
     band a.0,<$01.7

     band b.0,<$01.0
     band b.1,<$01.1
     band b.2,<$01.2
     band b.3,<$01.3
     band b.4,<$01.4
     band b.5,<$01.5
     band b.6,<$01.6
     band b.7,<$01.7
     band b.7,<$01.0
     band b.0,<$01.7

     biand a.0,<$01.7
     bor a.0,<$01.7
     bior a.0,<$01.7
     beor a.0,<$01.7
     bieor a.0,<$01.7
     ldbt a.0,<$01.7
     stbt a.0,<$01.7

     adcr a,b
     addr a,b
     andr a,b
     cmpr a,b
     eorr a,b
     orr  a,b
     sbcr a,b
     subr a,b

     tfm x+,y+
     tfm y-,x-
     tfm s+,u
     tfm y,u+

     leax e,x
     leax f,x
     leax e,y
     leax f,y
     leax e,s
     leax f,s
     leax e,u
     leax f,u

     leax [e,x]
     leax [f,x]
     leax [e,y]
     leax [f,y]
     leax [e,s]
     leax [f,s]
     leax [e,u]
     leax [f,u]

     leax w,x
     leax w,y
     leax w,s
     leax w,u

     leax [w,x]
     leax [w,y]
     leax [w,s]
     leax [w,u]

     leax ,w
     leax $8000,w
     leax ,w++
     leax ,--w

     leax [,w]
     leax [$8000,w]
     leax [,w++]
     leax [,--w]


