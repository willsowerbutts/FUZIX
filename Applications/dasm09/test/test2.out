; org $0000 
0000: 10 89 01 01    ADCD #$0101 
0004: 10 99 01       ADCD <$01 
0007: 10 A9 84       ADCD ,X 
000A: 10 B9 01 01    ADCD $0101 
000E: 11 8B 01       ADDE #$01 
0011: 11 9B 01       ADDE <$01 
0014: 11 AB 84       ADDE ,X 
0017: 11 BB 01 01    ADDE $0101 
001B: 11 CB 01       ADDF #$01 
001E: 11 DB 01       ADDF <$01 
0021: 11 EB 84       ADDF ,X 
0024: 11 FB 01 01    ADDF $0101 
0028: 10 8B 01 01    ADDW #$0101 
002C: 10 9B 01       ADDW <$01 
002F: 10 AB 84       ADDW ,X 
0032: 10 BB 01 01    ADDW $0101 
0036: 02 01 01       AIM #$01,<$01 
0039: 72 01 01 01    AIM #$01,$0101 
003D: 62 01 84       AIM #$01,,X 
0040: C3 01 01       ADDD #$0101 
0043: D3 01          ADDD <$01 
0045: E3 84          ADDD ,X 
0047: F3 01 01       ADDD $0101 
004A: 10 48          ASLD 
004C: 10 47          ASRD 
004E: 11 3C 01       BITMD #$01 
0051: 10 4F          CLRD 
0053: 11 4F          CLRE 
0055: 11 5F          CLRF 
0057: 10 5F          CLRW 
0059: 11 81 01       CMPE #$01 
005C: 11 91 01       CMPE <$01 
005F: 11 A1 84       CMPE ,X 
0062: 11 B1 01 01    CMPE $0101 
0066: 11 C1 01       CMPF #$01 
0069: 11 D1 01       CMPF <$01 
006C: 11 E1 84       CMPF ,X 
006F: 11 F1 01 01    CMPF $0101 
0073: 10 81 01 01    CMPW #$0101 
0077: 10 91 01       CMPW <$01 
007A: 10 A1 84       CMPW ,X 
007D: 10 B1 01 01    CMPW $0101 
0081: 10 43          COMD 
0083: 11 43          COME 
0085: 11 53          COMF 
0087: 10 53          COMW 
0089: 10 4A          DECD 
008B: 11 4A          DECE 
008D: 11 5A          DECF 
008F: 10 5A          DECW 
0091: 11 8D 01       DIVD #$01 
0094: 11 9D 01       DIVD <$01 
0097: 11 AD 84       DIVD ,X 
009A: 11 BD 01 01    DIVD $0101 
009E: 11 8E 01 01    DIVQ #$0101 
00A2: 11 9E 01       DIVQ <$01 
00A5: 11 AE 84       DIVQ ,X 
00A8: 11 BE 01 01    DIVQ $0101 
00AC: 05 01 01       EIM #$01,<$01 
00AF: 75 01 01 01    EIM #$01,$0101 
00B3: 65 01 84       EIM #$01,,X 
00B6: 10 88 01 01    EORD #$0101 
00BA: 10 98 01       EORD <$01 
00BD: 10 A8 84       EORD ,X 
00C0: 10 B8 01 01    EORD $0101 
00C4: 10 4C          INCD 
00C6: 11 4C          INCE 
00C8: 11 5C          INCF 
00CA: 10 5C          INCW 
00CC: 11 86 01       LDE #$01 
00CF: 11 96 01       LDE <$01 
00D2: 11 A6 84       LDE ,X 
00D5: 11 B6 01 01    LDE $0101 
00D9: 11 C6 01       LDF #$01 
00DC: 11 D6 01       LDF <$01 
00DF: 11 E6 84       LDF ,X 
00E2: 11 F6 01 01    LDF $0101 
00E6: CD 01 01 01 01 LDQ #$01010101 
00EB: 10 DC 01       LDQ <$01 
00EE: 10 EC 84       LDQ ,X 
00F1: 10 FC 01 01    LDQ $0101 
00F5: 10 86 01 01    LDW #$0101 
00F9: 10 96 01       LDW <$01 
00FC: 10 A6 84       LDW ,X 
00FF: 10 B6 01 01    LDW $0101 
0103: 11 3D 01       LDMD #$01 
0106: 10 44          LSRD 
0108: 10 54          LSRW 
010A: 11 8F 00 01    MULD #$0001 
010E: 11 9F 01       MULD <$01 
0111: 11 AF 84       MULD ,X 
0114: 11 BF 01 01    MULD $0101 
0118: 10 40          NEGD 
011A: 01 01 01       OIM #$01,<$01 
011D: 71 01 01 01    OIM #$01,$0101 
0121: 61 01 84       OIM #$01,,X 
0124: 10 8A 01 01    ORD #$0101 
0128: 10 9A 01       ORD <$01 
012B: 10 AA 84       ORD ,X 
012E: 10 BA 01 01    ORD $0101 
0132: 10 38          PSHSW 
0134: 10 3A          PSHUW 
0136: 10 39          PULSW 
0138: 10 3B          PULUW 
013A: 10 49          ROLD 
013C: 10 59          ROLW 
013E: 10 46          RORD 
0140: 10 56          RORW 
0142: 10 82 01 01    SBCD #$0101 
0146: 10 92 01       SBCD <$01 
0149: 10 A2 84       SBCD ,X 
014C: 10 B2 01 01    SBCD $0101 
0150: 14             SEXW 
0151: 11 97 01       STE <$01 
0154: 11 B7 01 01    STE $0101 
0158: 11 A7 84       STE ,X 
015B: 11 D7 01       STF <$01 
015E: 11 F7 01 01    STF $0101 
0162: 11 E7 84       STF ,X 
0165: 10 DD 01       STQ <$01 
0168: 10 FD 01 01    STQ $0101 
016C: 10 ED 84       STQ ,X 
016F: 10 97 01       STW <$01 
0172: 10 B7 01 01    STW $0101 
0176: 10 A7 84       STW ,X 
0179: 11 80 01       SUBE #$01 
017C: 11 90 01       SUBE <$01 
017F: 11 A0 84       SUBE ,X 
0182: 11 B0 01 01    SUBE $0101 
0186: 11 C0 01       SUBF #$01 
0189: 11 D0 01       SUBF <$01 
018C: 11 E0 84       SUBF ,X 
018F: 11 F0 01 01    SUBF $0101 
0193: 10 80 01 01    SUBW #$0101 
0197: 10 90 01       SUBW <$01 
019A: 10 A0 84       SUBW ,X 
019D: 10 B0 01 01    SUBW $0101 
01A1: 0B 01 01       TIM #$01,<$01 
01A4: 7B 01 01 01    TIM #$01,$0101 
01A8: 6B 01 84       TIM #$01,,X 
01AB: 10 4D          TSTD 
01AD: 11 4D          TSTE 
01AF: 11 5D          TSTF 
01B1: 10 5D          TSTW 
01B3: 11 30 00 01    BAND CC.0,<$01.0 
01B7: 11 30 09 01    BAND CC.1,<$01.1 
01BB: 11 30 12 01    BAND CC.2,<$01.2 
01BF: 11 30 1B 01    BAND CC.3,<$01.3 
01C3: 11 30 24 01    BAND CC.4,<$01.4 
01C7: 11 30 2D 01    BAND CC.5,<$01.5 
01CB: 11 30 36 01    BAND CC.6,<$01.6 
01CF: 11 30 3F 01    BAND CC.7,<$01.7 
01D3: 11 30 07 01    BAND CC.7,<$01.0 
01D7: 11 30 38 01    BAND CC.0,<$01.7 
01DB: 11 30 40 01    BAND A.0,<$01.0 
01DF: 11 30 49 01    BAND A.1,<$01.1 
01E3: 11 30 52 01    BAND A.2,<$01.2 
01E7: 11 30 5B 01    BAND A.3,<$01.3 
01EB: 11 30 64 01    BAND A.4,<$01.4 
01EF: 11 30 6D 01    BAND A.5,<$01.5 
01F3: 11 30 76 01    BAND A.6,<$01.6 
01F7: 11 30 7F 01    BAND A.7,<$01.7 
01FB: 11 30 47 01    BAND A.7,<$01.0 
01FF: 11 30 78 01    BAND A.0,<$01.7 
0203: 11 30 80 01    BAND B.0,<$01.0 
0207: 11 30 89 01    BAND B.1,<$01.1 
020B: 11 30 92 01    BAND B.2,<$01.2 
020F: 11 30 9B 01    BAND B.3,<$01.3 
0213: 11 30 A4 01    BAND B.4,<$01.4 
0217: 11 30 AD 01    BAND B.5,<$01.5 
021B: 11 30 B6 01    BAND B.6,<$01.6 
021F: 11 30 BF 01    BAND B.7,<$01.7 
0223: 11 30 87 01    BAND B.7,<$01.0 
0227: 11 30 B8 01    BAND B.0,<$01.7 
022B: 11 31 78 01    BIAND A.0,<$01.7 
022F: 11 32 78 01    BOR A.0,<$01.7 
0233: 11 33 78 01    BIOR A.0,<$01.7 
0237: 11 34 78 01    BEOR A.0,<$01.7 
023B: 11 35 78 01    BIEOR A.0,<$01.7 
023F: 11 36 78 01    LDBT A.0,<$01.7 
0243: 11 37 78 01    STBT A.0,<$01.7 
0247: 10 31 89       ADCR A,B 
024A: 10 30 89       ADDR A,B 
024D: 10 34 89       ANDR A,B 
0250: 10 37 89       CMPR A,B 
0253: 10 36 89       EORR A,B 
0256: 10 35 89       ORR A,B 
0259: 10 33 89       SBCR A,B 
025C: 10 32 89       SUBR A,B 
025F: 11 38 12       TFM X+,Y+ 
0262: 11 39 21       TFM Y-,X- 
0265: 11 3A 43       TFM S+,U 
0268: 11 3B 23       TFM Y,U+ 
026B: 30 87          LEAX E,X 
026D: 30 8A          LEAX F,X 
026F: 30 A7          LEAX E,Y 
0271: 30 AA          LEAX F,Y 
0273: 30 E7          LEAX E,S 
0275: 30 EA          LEAX F,S 
0277: 30 C7          LEAX E,U 
0279: 30 CA          LEAX F,U 
027B: 30 97          LEAX [E,X] 
027D: 30 9A          LEAX [F,X] 
027F: 30 B7          LEAX [E,Y] 
0281: 30 BA          LEAX [F,Y] 
0283: 30 F7          LEAX [E,S] 
0285: 30 FA          LEAX [F,S] 
0287: 30 D7          LEAX [E,U] 
0289: 30 DA          LEAX [F,U] 
028B: 30 8E          LEAX W,X 
028D: 30 AE          LEAX W,Y 
028F: 30 EE          LEAX W,S 
0291: 30 CE          LEAX W,U 
0293: 30 9E          LEAX [W,X] 
0295: 30 BE          LEAX [W,Y] 
0297: 30 FE          LEAX [W,S] 
0299: 30 DE          LEAX [W,U] 
029B: 30 8F          LEAX ,W 
029D: 30 AF 80 00    LEAX $8000,W 
02A1: 30 CF          LEAX ,W++ 
02A3: 30 EF          LEAX ,--W 
02A5: 30 90          LEAX [,W] 
02A7: 30 B0 80 00    LEAX [$8000,W] 
02AB: 30 D0          LEAX [,W++] 
02AD: 30 F0          LEAX [,--W] 
